OuterWildsRandomSpeedrun/OuterWildsRandomSpeedrun.cs

```diff
@@ -214,5 +222,36 @@ protected SpawnPoint GetSpawnPointByName(SpawnPoint[] spawnPoints, string name)
         {
             return spawnPoints.Where(point => { return point.name.Equals(name); }).First();
         }
+
+        protected void SpawnGoal(Transform parent)
+        {
+            var go = new GameObject("GoalPoint");
+            var collider = go.AddComponent<SphereCollider>();
+            collider.isTrigger = true;
+
+            var mesh = new GameObject("CollectibleMarshmellow_Mesh");
+            if (marshmallowMesh == null)
+                marshmallowMesh = GameObject.Find("Player_Body/RoastingSystem/Stick_Root/Stick_Pivot/Stick_Tip/Mallow_Root/Props_HEA_Marshmallow").GetComponent<MeshFilter>().mesh;
```

Comment:
can we set a var for the GameObject that's found so we can grab the two components we want from it without re-querying

Reply:
yes, smart